File: model/account_flags.go
package model

import (
  "github.com/stellar/go/xdr"
)

func NewAccountFlagsFromRaw(raw int, id string) AccountFlags {
  f := xdr.AccountFlags(raw)

	return AccountFlags{
    ID: id,
		AuthRequired:  f & xdr.AccountFlagsAuthRequiredFlag != 0,
		AuthRevokable: f & xdr.AccountFlagsAuthRevocableFlag != 0,
		AuthImmutable: f & xdr.AccountFlagsAuthImmutableFlag != 0,
	}
}

File: src/model2/account_flags.ts
export interface IAccountFlags {
  authRequired: boolean;
  authRevokable: boolean;
  authImmutable: boolean;
}

export class AccountFlags implements IAccountFlags {
  public authRequired: boolean;
  public authRevokable: boolean;
  public authImmutable: boolean;

  constructor(data: IAccountFlags) {
    this.authRequired = data.authRequired;
    this.authRevokable = data.authRevokable;
    this.authImmutable = data.authImmutable;
  }

  public equals(other: IAccountFlags): boolean {
    return (
      this.authRequired === other.authRequired &&
      this.authRevokable === other.authRevokable &&
      this.authImmutable === other.authImmutable
    );
  }
}
